src/arduino-timer.h

```diff
@@ -94,7 +94,7 @@ class Timer {
         if (!task) return;
 
         timer_foreach_task(t) {
-            if (t->handler && (t->id ^ task) == (uintptr_t)t) {
```

Comment:
The `uintptr_t` cast is preferred here. The result of the xor doesn't represent a `Task`, but a conversion from a `Task` to a numeric value, in this case a pointer value.

At the moment `Task` is defined as `uintptr_t`, but that might not always be the case, hence the direct cast to `uintptr_t`.